code/modules/mob/living/carbon/human/death.dm

```diff
@@ -75,14 +75,22 @@
 	// Finding the last guy for anti-delay.
 	if(SSticker.mode && SSticker.mode.is_in_endgame && SSticker.current_state != GAME_STATE_FINISHED && is_mainship_level(z))
 		var/mob/last_living_human
+		var/shipside_humans_count = 0
+		var/datum/hive_status/main_hive = GLOB.hive_datum[XENO_HIVE_NORMAL]
+		var/see_humans_on_tacmap = main_hive.see_humans_on_tacmap
 		for(var/mob/living/carbon/human/cur_human as anything in GLOB.alive_human_list)
 			if(!is_mainship_level(cur_human.z))
 				continue
-			if(last_living_human)
+			shipside_humans_count++
+			if(last_living_human && see_humans_on_tacmap)
 				last_living_human = null
 				break
 			last_living_human = cur_human
-		if(last_living_human && (GLOB.last_qm_callout + 2 MINUTES) < world.time)
+
+		if(!see_humans_on_tacmap && shipside_humans_count < length(main_hive.totalXenos) * 0.8)
```

Comment:
This change will require #5122 but make the ratio (0.8) be a define like in that PR near the other defines. Also count the xenos using get_real_total_xeno_count() from that PR as well.